lib/engine/share_price.rb

```diff
@@ -107,5 +110,31 @@ def normal_movement?
       # Can be moved into normally, rather than something custom such as not owning a train.
       @type != :liquidation
     end
+
+    def report_last_move
+      r0, c0 = @last_reported
+      r1, c1 = @coordinates
+
+      return nil if @limit_hit.nil? && r0 == r1 && c0 == c1
+
+      msg = ''
+
+      if r0 != r1 || c0 != c1
+        msg += 'moves '
+        if r0 != r1
+          msg += "#{(r1 - r0).abs} #{r1 < r0 ? 'up' : 'down'} "
+        end
+        if c0 != c1
+          msg += 'and ' if r0 != r1
+          msg += "#{(c1 - c0).abs} #{c1 < c0 ? 'left' : 'right'} "
+        end
+      elsif @limit_hit
+        msg += "is #{@limit_hit == :ledge ? 'ledged' : 'at ceiling'} "
+      end
+
+      @last_reported = @coordinates
```

Comment:
Since the last reported coordinates are being set in the stock market move method, why does it need to be set here? I am concerned that having a report method change state will produce unpredictable behavior in edge cases.

Reply:
Following @roseundy's suggestion I'm working on a better version that modifies log_share_price to compare two share_price objects instead of keeping track of last_reported coordinates, so hopefully this issue won't appear.